### PR TITLE
Add TESTING.md and add WIKI page

### PR DESCRIPTION
**Pull request by davidzwa**
- [x] Add TESTING.md
- [x] Refer to TESTING.md in README.md
- [ ] Get issue approved
- [ ] Add `Developing Tests` sub-page to `Developer Guides` WIKI page.

### CODE REVIEW

**Comment by davidzwa** [comment]
Sorry for the weird commit history (fork was outdated). Changes are only on commits after 19c3ad7 and beyond.

Fixes #8659

File: README.md [hunk 1]
```diff
@@ -239,8 +239,7 @@ The basic way to run tests:
     $ python2 -m pip install -U typing
     $ ./runtests.py
 
-For more on the tests, such as how to write tests and how to control
-which tests to run, see [Test README.md](test-data/unit/README.md).
+For more on the tests and how to develop and run test locally, please refer to [Test README.md](TESTING.md) for a complete overview.
```
Comment (gvanrossum) [inline README.md:242]: Please break this line around column 72 (like done elsewhere in the file).

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# TESTING mypy
+The basic way to run all tests:
+
+    $ pip3 install -r test-requirements.txt
+    $ python2 -m pip install -U typing
+    $ ./runtests.py
+
+Mypy is tested using `pytest`. This can be installed as shown above, using `test-requirements.txt`. In essence, there are multiple kinds of tests for mypy:
```
Comment (gvanrossum) [inline TESTING.md:8]: Again, please break all lines longer than 72 characters (that's a soft limit, but they can't be longer than 79 characters unless there's a single very long URL).

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- Unit tests (Found under `./mypy/test/`, [example](mypy/test/testapi.py))
+- Data-driven unit and integration tests (Found under `./mypy/test/`, [example pep561](mypy/test/testpep561.py))
+
+As might be clear, some tests are driven by data. For now, this data resides in `./test-data/unit/`. Do note: this syntax of these data-driven tests is parsed by Python test-cases in `./mypy/test/` by specifying the `files=[]` property. Some test files specify their own conditions on the tests (input data, flags, etc.). For more information about the syntax, read the next section.
```
Comment (gvanrossum) [inline TESTING.md:12]: I'd drop the "For now," -- I don't think anyone is thinking about moving it elsewhere.

Also:

- "this syntax" -> "the syntax"
- "specifying the `files=[]` property" -> "setting the `files` class attribute to a list of filenames"
- I don't actually see anything in the next section about the syntax of the test data files; it basically just contains a link. Maybe you can drop that section and just plant the link here?

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+Such a data-driven test can look like:
+
+```
+class PEP561Suite(DataSuite):
+    files = ['pep561.test'] # Specify the list of test data files
```
Comment (gvanrossum) [inline TESTING.md:18]: I would clarify that this reads tests from `test-data/unit/pep561.test`.

**Pull request closed without merging**